### ISSUE TITLE
Suppress errors from sensors on each prompt

### ISSUE THREAD

**Issue by jnweiger** [nojhan/liquidprompt#598]
Shell: bash
Operating system: Linux Mint Tricia, kernel  5.3.0-42
Liquid Prompt version (tag, commit): latest master

Seen after an online update to kernel 5.3.0-42 on Linux Mint Tricia.
With this kernel, /usr/bin/sensors -u babbles on stderr, as - yes - my
/sys/class/hwmon/hwmon4/temp2_input is not readable. All other temp*_input are readable.

I am well aware ethat this was reported earlier in e.g. 
#534, #445, #394 and was turned down with the the argument that sensors should be fixed.

While I fully agree, that the correct fix goes elsewhere, I'd still argue, we should ease the symptoms within liquidprompt.

As part of liquidprompt the output is not helpful. Users have no context. (the command that fails is not mentioned, the /sys/ node that fails is not mentioned). Users get spammed with the error. (seen on every prompt)

Here is a patch against develop. Please reconsider: https://github.com/nojhan/liquidprompt/pull/597

**Comment by Rycieos** [comment]
Can you paste some of the output you are getting spammed with?

> I'd still argue, we should ease the symptoms within liquidprompt.
As part of liquidprompt the output is not helpful. Users have no context. 

I agree that the error messages should be helpful. If they aren't (again, need to see what they actually are), then we should fix them to be more helpful. Your suggested patch, however, is not more helpful; it is less. It suppresses error output, so users won't see at all that there is a problem, much less know what it is.

**Comment by jnweiger** [comment]
If or if not /usr/bin/sensors reports this as an error or not is inconsistent.

`/usr/bin/sensors` (without parameters) nicely says N/A for the missing value
```coretemp-isa-0000
Adapter: ISA adapter
Package id 0:  +49.0°C  (high = +100.0°C, crit = +100.0°C)
Core 0:        +46.0°C  (high = +100.0°C, crit = +100.0°C)
Core 1:        +46.0°C  (high = +100.0°C, crit = +100.0°C)
Core 2:        +48.0°C  (high = +100.0°C, crit = +100.0°C)
Core 3:        +47.0°C  (high = +100.0°C, crit = +100.0°C)

thinkpad-isa-0000
Adapter: ISA adapter
fan1:           0 RPM
temp1:        +46.0°C  
temp2:            N/A  
temp3:         +0.0°C  
temp4:         +0.0°C  
temp5:         +0.0°C  
temp6:         +0.0°C  
temp7:         +0.0°C  
temp8:         +0.0°C  
temp9:         +0.0°C  
temp10:        +0.0°C  
temp11:       +66.0°C  
temp12:        +0.0°C  
temp13:        +0.0°C  
temp14:        +0.0°C  
temp15:        +0.0°C  
temp16:        +0.0°C  
```

but `/usr/bin/sensors -u` produces an error:
```
...
thinkpad-isa-0000
Adapter: ISA adapter
fan1:
  fan1_input: 0.000
temp1:
  temp1_input: 46.000
temp2:
ERROR: Can't get value of subfeature temp2_input: Can't read
temp3:
  temp3_input: 0.000
temp4:
  temp4_input: 0.000
temp5:
  temp5_input: 0.000
temp6:
...
```

**Comment by jnweiger** [comment]
Example shell session with "spamming":
```
θ66° [testy:~] $ pwd
/home/testy
ERROR: Can't get value of subfeature temp2_input: Can't read
θ66° [testy:~] $ cd t
ERROR: Can't get value of subfeature temp2_input: Can't read
θ66° [testy:~/t] $ ls
 04_bouncing_fraph.bin                                 bouncing_graph.mp4          input                     wooden_toy_train-b2x2.mp4
 05_spinning_heart.bin                                 fablabnbg_logo-b0x2.mp4     rotating_cube.bin
ERROR: Can't get value of subfeature temp2_input: Can't read
θ66° [testy:~/t] $ cd ..
ERROR: Can't get value of subfeature temp2_input: Can't read
θ66° [testy:~] $ 
```

**Comment by Rmano** [comment]
I have (or had, didn't test anymore) the same problem, I use this patch:
```diff
diff --git a/liquidprompt b/liquidprompt
index 190ab8f..1860105 100755
--- a/liquidprompt
+++ b/liquidprompt
@@ -1514,7 +1514,7 @@ if (( LP_ENABLE_TEMP )); then
         # Return the hottest system temperature we get through the sensors command
         # Only the integer part is retained
         local -i i
-        for i in $(sensors -u |
+        for i in $(sensors -u 2> /dev/null|
                 sed -n 's/^  temp[0-9][0-9]*_input: \([0-9]*\)\..*$/\1/p'); do
             (( $i > ${temperature:-0} )) && temperature=i
         done
```
which works ok for me.

**Comment by Rycieos** [comment]
Fixed in 4a526965cba546978423a4d51bfbb0a2d1000246. See the commit message for the full details.

I apologize, I was an idiot on this one.
